src/Types.cg.fs

```diff
@@ -1761,17 +1825,88 @@ type DidCloseNotebookDocumentParams =
+        /// requested.
+        Context: InlineCompletionContext
+    }
+
+    interface ITextDocumentPositionParams with
+        /// The text document.
+        member x.TextDocument = x.TextDocument
+        /// The position inside the text document.
+        member x.Position = x.Position
+
+    interface IWorkDoneProgressParams with
+        /// An optional token that a server can use to report work done progress.
+        member x.WorkDoneToken = x.WorkDoneToken
 
-type _InitializeParamsClientInfo =
+/// Represents a collection of {@link InlineCompletionItem inline completion items} to be presented in the editor.
+///
+/// @since 3.18.0
+/// @proposed
+type InlineCompletionList =
     {
-        /// The name of the client as defined by the client.
-        Name: string
-        /// The client's version as defined by the client.
-        Version: string option
+        /// The inline completion items
+        Items: InlineCompletionItem[]
+    }
+
+/// An inline completion item represents a text snippet that is proposed inline to complete text that is being typed.
+///
+/// @since 3.18.0
+/// @proposed
+type InlineCompletionItem =
+    {
+        /// The text to replace the range with. Must be set.
+        InsertText: U2<string, StringValue>
+        /// A text that is used to decide if this inline completion should be shown. When `falsy` the {@link InlineCompletionItem.insertText} is used.
+        FilterText: string option
+        /// The range to replace. Must begin and end on the same line.
+        Range: Range option
+        /// An optional {@link Command} that is executed *after* inserting this completion.
+        Command: Command option
+    }
+
+/// Inline completion options used during static or dynamic registration.
+///
+/// @since 3.18.0
+/// @proposed
+type InlineCompletionRegistrationOptions =
+    {
+        WorkDoneProgress: bool option
+        /// A document selector to identify the scope of the registration. If set to null
+        /// the document selector provided on the client side will be used.
+        [<JsonProperty(NullValueHandling = NullValueHandling.Include)>]
+        DocumentSelector: DocumentSelector option
+        /// The id used to register the request. The id can be used to deregister
+        /// the request again. See also Registration#id.
+        Id: string option
     }
 
+    interface IInlineCompletionOptions
+
+    interface IWorkDoneProgressOptions with
+        member x.WorkDoneProgress = x.WorkDoneProgress
+
+    interface ITextDocumentRegistrationOptions with
+        /// A document selector to identify the scope of the registration. If set to null
+        /// the document selector provided on the client side will be used.
+        member x.DocumentSelector = x.DocumentSelector
+
+type RegistrationParams = { Registrations: Registration[] }
+type UnregistrationParams = { Unregisterations: Unregistration[] }
```

Comment:
typo in the metamodel here maybe? `Unregisterations` isn't a word, `Unregistrations` would be (ish).

Reply:
// This should correctly be named `unregistrations`. However, changing this
	// is a breaking change and needs to wait until we deliver a 4.x version
	// of the specification.
	
	https://microsoft.github.io/language-server-protocol/specifications/lsp/3.18/specification/#unregistrationParams

Reply:
that's amazing :D